### ISSUE TITLE
Users cannot add video metadata in the visual editor

### ISSUE THREAD

**Issue by karlgroves** [WordPress/gutenberg#15281]
Users cannot add video metadata in the visual editor

-   **Severity:** Medium
-   **Affected Populations:**
    -   Blind
    -   Low-Vision
    -   Cognitively Impaired
-   **Platform(s):**
    -   All / Universal
-   **Components affected:**
    -   Block Types
    -   Output Markup

#### Issue description

Authors cannot add captions (such as .srt or .vtt files) to video media
using the visual editor.

Users who do not have HTML knowledge may not know how to do these things
using HTML, resulting in them publishing content which is less
accessible to readers.

#### Remediation Guidance
The ability to add captions should be part of the Video block's visual
editor, similar to the way that image media provides the option to add a
caption.

#### Relevant standards

-   [1.2.2 Captions (Prerecorded)
    (Level A)](https://www.w3.org/TR/WCAG20/#media-equiv-captions)

**Note:** This issue may be a duplicate with other existing accessibility-related bugs in this project. This issue comes from the Gutenberg accessibility audit, performed by [Tenon](https://www.tenon.io) and funded by [WP Campus](https://wpcampus.org/). This issue is GUT-113 in Tenon's report

**Comment by mapk** [comment]
> The ability to add captions should be part of the Video block's visual
> editor, similar to the way that image media provides the option to add a
> caption.

After testing in Gutenberg 5.6, this appears to be resolved. I'd like this to be verified before I close this. I'm not entirely sure this is what's being referenced in this issue because it mentions .srt and .vtt files which seem to allude to closed captions. Can this be clarified?

**Example:**

![video](https://user-images.githubusercontent.com/617986/57032206-aecc4a80-6bfe-11e9-9f66-b1f378168425.gif)

**Comment by brentswisher** [comment]
Just a note here to try and be helpful. The "caption" referenced here has a specific definition for accessibility.
[https://www.w3.org/TR/WCAG20/#captionsdef](https://www.w3.org/TR/WCAG20/#captionsdef)

Basically, a caption is a text track that displays what is currently happening/heard a certain point in a video. These can be:

- Open Captions - Text is part of the video itself and always displayed. You will probably have seen these in Facebook videos/ads that expect the audio to be muted. 
- Closed Captions - Text that can be turned on/off like you may be familiar with from movies or television shows. 

For Gutenberg, if it is a video that is converted to an embed (youtube, vimeo, etc.) the embed source would be providing this. If it is video file though, to pass WCAG 1.2.2 it should provide a way to upload a caption track and enable it in the HTML player. [Mozilla](https://developer.mozilla.org/en-US/docs/Web/Guide/Audio_and_video_delivery/Adding_captions_and_subtitles_to_HTML5_video) has a good page explaining the HTML to do this.

**Comment by afercia** [comment]
@mapk yes this related to closed captions. In the Classic Editor it is possible to add closed captions (and alternate sources) to videos so this is lack of parity with Gutenberg and the Classic Editor.

Current Classic Editor:
<img width="861" alt="bildschirmfoto 2018-07-03 um 00 07 22" src="https://user-images.githubusercontent.com/695201/42188636-1e166864-7e55-11e8-9a70-e052f22003f4.png">

There are already issues to keep _track_ of captions and sources:
#7673 for subtitle / closed captions 
#9457 for alternate sources

**Comment by melchoyce** [comment]
> There are already issues to keep _track_ of captions and sources:

😆 

Closing in favor of #7673 and #9457.
